server_extensions_api/src/main/java/com/generalbytes/batm/server/extensions/IExtensionContext.java

```diff
@@ -320,12 +321,24 @@ IIdentity updateIdentity(String identityId, String externalId, int state, int ty
     ITransactionSellInfo sellCrypto(String terminalSerialNumber, BigDecimal fiatAmount, String fiatCurrency, BigDecimal cryptoAmount, String cryptoCurrency, String identityPublicId, String discountCode) throws SellException;
 
     /**
-     * Call this transaction to create a cash back transaction. After this call server will allocate cash for the customer that can visit machine and withdraw cash.
+     * Create a buy transaction (sends crypto to the provided destination address)
+     * @param terminalSerialNumber
      * @param fiatAmount
      * @param fiatCurrency
-     * @param identityPublicId
-     * @return - read ITransactionSellInfo.getTransactionUUID() to find out what should be filled in sell QR code.
-     */
+     * @param cryptoAmount
+     * @param cryptoCurrency
+     * @param destinationAddress
+     * @return
+     * @throws BuyException
+     */
+    ITransactionBuyInfo buyCrypto(String terminalSerialNumber, BigDecimal fiatAmount, String fiatCurrency, BigDecimal cryptoAmount, String cryptoCurrency, String destinationAddress) throws BuyException;
```

Comment:
Add also following parameters identityPublicId and discountCode.

Reply:
added